addons/portal/controllers/web.py

```diff
@@ -21,6 +21,6 @@ def _login_redirect(self, uid, redirect=None):
 
     @http.route()
     def web_client(self, s_action=None, **kw):
-        if request.session.uid and not request.env['res.users'].sudo().browse(request.session.uid).has_group('base.group_user'):
+        if 'redirect' not in kw and request.session.uid and not request.env['res.users'].sudo().browse(request.session.uid).has_group('base.group_user'):
```

Comment:
Not sure it really matters as this is used to avoid intercepting baseline behaviour, so worst case scenario this would do something like let the parent method redirect to `request.redirect('/web/login?error=access')`.

Still, `dict.get` does match the behaviour / filter of the parent method so seems good enough